### ISSUE TITLE
deprecate `groupby` and rename to `group_by`

### ISSUE THREAD

**Issue by ritchie46** [pola-rs/polars#10522]
### Problem description

It is becoming a more and more glaring inconsistency.

**Comment by stinodego** [comment]
It makes sense at this point. We can keep `groupby` as an alias for a while. `group_by` will be first in alphabetical order so it will be easy to autocomplete the right option.

**Comment by ritchie46** [comment]
Yes, let's introduce it slowly. This can take a few release cycles.

**Comment by deanm0000** [comment]
:(

**Comment by avimallu** [comment]
I agree that while learning Polars, I found this unusual and had to correct to ’groupby’ every single time. One thing to keep consistent is to change the user guide the same time as when it is aliased.

**Comment by mcrumiller** [comment]
I generally prefer the underscore approach to be consistent when a function uses multiple words, but the term `groupby` has become so prevalent that it itself is basically a single word at this point. Wolfram even has a general [definition for it](https://reference.wolfram.com/language/ref/GroupBy.html.en).

**Comment by stinodego** [comment]
You can see from their definition (`GroupBy`) though that it consists of two words. SQL is the same (`GROUP BY`), and Spark as well (`groupBy`).

We would be doing the same, just using snake case (`group_by`).

**Comment by mcrumiller** [comment]
Alright, I'm sold.

**Comment by stinodego** [comment]
If no strong objections pop up here, I'll implement this sometime next week. I'd like this to be part of `0.19`.
